### PR TITLE
feat: issue #32 prev and next tree button

### LINKED ISSUE #32: Add feature: previous tree and next tree button
Screen Shot 2021-11-04 at 11 05 06 AM

Actually, the function of this feature has been finished in the https://github.com/Greenstand/treetracker-web-map-client/tree/master also you can find it in the website online : map.treetracker.org

The goal of this issue is to:
* Migrate this feature to the `core`
* Temporarily add two button (as shown in the picture) and add the logic of clicking these two buttons. 
* We will open a new issue when our designer has finished the design of the buttons, and replace the UI with new design.

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-core#treetracker-web-map-core) for more information/guide/tutorial.
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.

### PR DESCRIPTION
**Pull request by VWRoli**
The purpose of this PR is to fix #32

### CODE REVIEW

**Comment by dadiorchen** [comment]
@VWRoli how about the mobile version for this feature?

File: src/Map.js [hunk 1]
```diff
@@ -141,19 +142,41 @@ export default class Map {
...
+          log.warn('go next failed', e);
+        }
+      } else if (e.target.id === 'left-arrow') {
+        try {
+          this.goPrevPoint();
+        } catch (e) {
+          log.warn('go prev failed', e);
+        }
+      }
+    });
```
Comment (dadiorchen) [inline src/Map.js:179]: Do you think it is good that we move these lines to the button panel and expose: onClickNext, onClickPrev,

Comment (dadiorchen) [reply src/Map.js:179]: I'm considering add an option to `disablePrevNextButton`

**Comment by VWRoli** [comment]
> @VWRoli how about the mobile version for this feature?

What do you mean exactly?

File: src/Map.js (hunk 1 above)
Comment (VWRoli) [reply src/Map.js:179]: I'm not sure what you mean. The way I did it exposes these functions? How should I go about not exposing them? This repo is a bit advanced for me, just started working on it for fun and for the challenge.

Comment (dadiorchen) [reply src/Map.js:179]: @VWRoli no no you are doing great, I just want that you move the code above into the ButtonPanel, so the behavior of it gets encapsulated into the ButtonPanel component, I think it makes more sense, for example, I open a new option called: 
new Map({disableButtonPanel})

then, in the current code, we need to take care of two places:
No1.
if(dsableButtonPanel){
    new ButtonPanel()...
}

No.2:
if(disableButtonPanel){
  buttonPanelTarget.addListner(....)
}

So it's getting cumbersome, in my mind, hiding the details (encapsulation) is the major benefit of using the component like ButtonPanel

**Comment by dadiorchen** [comment]
> > @VWRoli how about the mobile version for this feature?
> 
> What do you mean exactly? Should I place them somewhere else on small screens? But this whole app does not have any mobile friendliness.

@VWRoli yes, let see what happens, let's forget it now, we can open a new issue on this if needed. so, please ignore this.

File: src/Map.js (hunk 1 above)
Comment (dadiorchen) [reply src/Map.js:179]: oh, maybe you need to expose events on the ButtonPanel:
new ButtonPanel({
onNext: this.goNextPoint,
onPrev: this.goPrevPoint,
})

Comment (VWRoli) [reply src/Map.js:179]: I understand now. I will see what I can do about it.

**Comment by dadiorchen** [comment]
@VWRoli here is the design of these two arrow button, can you implement it? https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap?node-id=6133%3A21294

**Comment by VWRoli** [comment]
Yes, sure.

**Comment by VWRoli** [comment]
Screenshot from 2022-01-16 10-57-03
Implemented the buttons, and tried to encapsulate the code into the buttonPanel component. Without much success. I get the following error message:
Screenshot from 2022-01-16 10-50-58
The correct functions are getting called but payload in `goNextPoint` is undefined. Which is the same error we get when we use the `goNextPoint` method without selecting a Marker. So the issue is somewhere with selecting the marker and encapsulating the code. I just can't figure out what exactly the problem is.
https://github.com/Greenstand/treetracker-web-map-core/blob/41f60409ff34b9cf2590932461263a2f7d93338b/src/Map.js#L830-L837

Could you take a look at it @dadiorchen?

**Comment by dadiorchen** [comment]
Good, thanks for the progress, yes, I will look into it, and I will back to you when I get any progress.

**Comment by dadiorchen** [comment]
@VWRoli thanks for you meaningful work on this, I finished this feature based on you work, take a look here: https://github.com/Greenstand/treetracker-web-map-core/pull/80/files

**Pull request closed without merging**